src/app/view/components/dot-theme-selector-dropdown/dot-theme-selector-dropdown.component.ts

```diff
@@ -0,0 +1,135 @@
+        this.paginatorService.paginationPerPage = 5;
+
+        this.siteService
+            .getCurrentSite()
+            .pipe(
+                take(1),
+                pluck('identifier'),
+                mergeMap((identifier) => {
+                    this.currentSiteIdentifier = identifier;
+                    this.paginatorService.setExtraParams('hostId', identifier);
+                    return this.paginatorService.getWithOffset(0).pipe(take(1));
+                })
+            )
+            .subscribe((themes) => {
+                this.themes = themes;
+                this.totalRecords = this.paginatorService.totalRecords;
+            });
+    }
+
+    propagateChange = (_: any) => {};
+    registerOnTouched(): void {}
+
+    /**
+     * Set the function to be called when the control receives a change event.
+     * @param any fn
+     * @memberof SearchableDropdownComponent
+     */
+    registerOnChange(fn): void {
+        this.propagateChange = fn;
+    }
+
+    /**
+     * Writes a new value to the element
+     *
+     * @param {DotTheme} theme
+     * @memberof DotThemeSelectorDropdownComponent
+     */
+    writeValue(theme: DotTheme): void {
+        this.value = theme;
+    }
+
+    /**
+     * Handles the onChange behavior of the select input
+     *
+     * @param {DotTheme} { identifier }
+     * @memberof DotThemeSelectorDropdownComponent
+     */
+    onChange({ identifier }: DotTheme) {
+        this.themesService
+            .get(identifier)
+            .pipe(take(1))
+            .subscribe((theme) => {
+                this.selectedTheme = theme.name;
+                this.writeValue(theme);
+            });
+        this.propagateChange(identifier);
+    }
+    /**
+     * Fetches a list of themes using an offset for pagination
+     *
+     * @param {LazyLoadEvent} event
+     * @return void
+     * @memberof DotThemeSelectorDropdownComponent
+     */
+    handlePageChange(event: LazyLoadEvent): void {
+        if (!this.currentSiteIdentifier) return;
+
+        this.currentOffset = event.first;
+
+        this.paginatorService
+            .getWithOffset(event.first)
+            .pipe(take(1))
+            .subscribe((themes) => {
+                this.themes = themes;
+            });
+    }
+    /**
+     *  Fetch theme list via the DotThemeSelectorDropdownComponent input text
+     *
+     * @param {string} filter
+     * @memberof DotThemeSelectorDropdownComponent
+     */
+    handleFilterChange(filter: string): void {
+        this.getFilteredThemes(filter);
+    }
+
+    private getFilteredThemes(filter = '', offset = 0): void {
+        this.paginatorService.searchParam = filter;
+        this.paginatorService
+            .getWithOffset(this.currentOffset ? this.currentOffset : offset)
+            .pipe(take(1))
+            .subscribe((themes) => {
+                this.themes = themes;
+                this.totalRecords = this.paginatorService.totalRecords;
```

Comment:
If you updates the `totalRecords` here, shouldn't you updated again in `handlePageChange`?

Reply:
@fmontes No, `this.totalRecords` is set initially in `ngOnInit` and then in `getFilteredThemes` — right? so, when the results are filtered it runs again and sets `totalRecords` to the returned value from `this.paginatorService.totalRecords` and when you clear the input it runs again setting it to its initial value.